src/org/authInfo.ts

```diff
@@ -419,7 +419,48 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
             logger.debug(`error updating auth file for ${orgAuthInfo.getUsername()}`, error);
           }
         } catch (error) {
-          logger.error(`Error connecting to devhub ${hubAuthInfo.username}`, error);
+          if (error instanceof Error && error.name === 'SingleRecordQuery_NoRecords' && fields.orgId) {
```

Comment:
line 395, the `getDevHubAuthInfos` is going to filter out orgs that aren't devHubs.  We'd want a better function for filtering out anything we know is a scratch org or sandbox, and then check the remaining stuff.

The isDevHub is a good filter to making the list of orgs we'd want to ask about scratch orgs

Reply:
Making those changes.